routing/grandcentral/server.go

```diff
@@ -42,91 +42,53 @@ func (s *Server) HandleRequest(ctx context.Context, p peer.ID, req *dhtpb.Messag
 func (s *Server) handleMessage(
 	ctx context.Context, p peer.ID, req *dhtpb.Message) (peer.ID, *dhtpb.Message) {
 
-	//  FIXME threw everything into this switch statement to get things going.
-	//  Once each operation is well-defined, extract pluggable backend so any
-	//  database may be used.
+	log.EventBegin(ctx, "routingMessageReceived", req, p, s.local).Done() // TODO may need to differentiate between local and remote
 
 	var response = dhtpb.NewMessage(req.GetType(), req.GetKey(), req.GetClusterLevel())
 	switch req.GetType() {
 
 	case dhtpb.Message_GET_VALUE:
-		dskey := util.Key(req.GetKey()).DsKey()
-		val, err := s.datastore.Get(dskey)
+		rawRecord, err := getRoutingRecord(s.routingBackend, util.Key(req.GetKey()))
 		if err != nil {
-			log.Error(errors.Wrap(err))
-			return "", nil
-		}
-		rawRecord, ok := val.([]byte)
-		if !ok {
-			log.Errorf("datastore had non byte-slice value for %v", dskey)
-			return "", nil
-		}
-		if err := proto.Unmarshal(rawRecord, response.Record); err != nil {
-			log.Error("failed to unmarshal dht record from datastore")
 			return "", nil
 		}
+		response.Record = rawRecord
 		// TODO before merging: if we know any providers for the requested value, return those.
```

Comment:
I didn't realize I'd left behind so many breadcrumbs. I'll re-read, address outstanding self-TODOs and ping for CR afterward.